babyai/bot.py

```diff
@@ -333,10 +350,20 @@ def get_action(self):
                 # otherwise: forward, left and right are full, we assume that you can go behind
                 return self.actions.left
 
-            # If we are right in front of the object, then the subgoal is completed
-            # go back to the previous subgoal
-            if np.array_equal(obj_pos, self.fwd_pos):
-                return self.subgoal_accomplished()
+            if not self.adjacent:
```

Comment:
A bit of nit-picking: `if self.adjacent` is better than `if not self.adjacent`:

https://www.python.org/dev/peps/pep-0008/#programming-recommendations